documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,231 @@
+
+>    MUST ensure that the prompt parameter contains consent unless other conditions for processing the request permitting offline access to the requested resources are in place; unless one or both of these conditions are fulfilled, then it MUST ignore the offline_access request,
+
+>    MUST ignore the offline_access request unless the Client is using a response_type value that would result in an Authorization Code being returned,
+
+>    MUST explicitly receive or have consent for offline access when the registered application_type is web,
+
+>    SHOULD explicitly receive or have consent for offline access when the registered application_type is native.
+
+The OIDC section on [OIDC Using Refresh Tokens](https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens) is not changed by this document.
+
+The OIDC security and privacy considerations regarding offline access and refresh tokens apply e.g. on
+[Token lifetime](https://openid.net/specs/openid-connect-core-1_0.html#TokenLifetime) and [Offline Access Privacy](https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccessPrivacy).
+
+## Client-Initiated Backchannel Authentication Flow
+
+The [Client-Initiated Backchannel Authentication (CIBA)](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html) flow is used to initiate an out-of-band authentication of the end-user.
+
+Communication with the Backchannel Authentication Endpoint MUST utilize TLS.
+
+CIBA allows the Client to get the authentication result in three ways: poll, ping, or push. This profile allows clients to use the *poll* mode. In the Poll mode, the authentication result is retrieved by the Client by polling the OP's token endpoint using the new grant type. 
+
+### Authentication Request
+
+This CAMARA document makes the login_hint parameter REQUIRED if CIBA is used in a CAMARA context.
+
+This CAMARA document clarifies the values used in login_hint in the following way:
+
+    * **_tel_**
+
+      For phone numbers. The `login_hint` must be a tel URI as defined in [RFC 3966](https://www.rfc-editor.org/info/rfc3966) for global phone numbers without visual separators in [E.164](https://www.itu.int/rec/T-REC-E.164-201011-I/en) format. For example, `tel:+34666666666`.
+
+    * **_ipport_**
+
+      For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.
+
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
+
+## Client Credentials Flow
+
+The [OAuth 2.0 Client Credentials](https://datatracker.ietf.org/doc/html/rfc6749#section-4.4) grant type is used to obtain a 2-legged Access Token that does not represent a user. This grant type can only be used when no personal user data is processed, and it is only a valid option to access the CAMARA APIs for these specific scenarios.
```

Comment:
I think that this [PR](https://github.com/camaraproject/IdentityAndConsentManagement/pull/120/files) enabled Camara API services sub-projects to allow some specific API invokers to consume APIs based on client credentials. So it looks that for some specific clients, CSP can apply client credentials flow, even for API services that results in sharing end-user characteristics, e.g. SimSwap or KYC Match.

Reply:
I think that what the Identity & Consent Working Group mandates regarding client credentials is 100% clear.  **This permission type can only be used when no personal user data is processed, and it is only a valid way to access the CAMARA APIs for those specific scenarios**. It has been clarified many times, and this policy applies to the CAMARA API subproject as a commonality.

https://github.com/camaraproject/IdentityAndConsentManagement/pull/120/files was NOT intended to change the working group's criteria for when client credentials can be used. So trying to misinterpret this text in order to circumvent the working group's guidelines is not the way to go.

CC @AxelNennker @Elisabeth-Ericsson

Reply:
In fact, there is a clear [documented guideline](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#mandatory-template-for-infodescription-in-camara-api-specs) as well for all CAMARA API subprojects to adopt ASAP to be aligned with WG to include in info.description property in the CAMARA API specs:

--

>### Authorization and authentication
>
>CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-API-access-and-user-consent.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md). Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
>
>**It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.**

Reply:
@jpengar @DT-DawidWroblewski 

Ee should be careful not to introduce different statements within the different documents. The agreed guidelines says (highlighting by me):

> It is important to remark that in cases where personal user data is processed by the API, **and** users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory

Which means that there could be cases, where personal user data is processed, but users have no options to exercise (e.g. if the processing is required by law). In these cases 3-legged might be not mandatory.

But the sentence above in the PR

> This grant type can only be used when no personal user data is processed, and it is only a valid option to access the CAMARA APIs for these specific scenarios.

would exclude such cases. It also misses the reference to "being subject to the prevailing legal framework dictated by local legislation".

My recommendation is to not discuss the use cases of valid client credentials here again, but delete it here and add a reference to the agreed guideline.

Reply:
An alternative could be, to copy the full and correct statement. But not a shortened one which is changing the semantic.

Reply:
Yes @hdamker. The agreed condition for the use of client credentials grant type is more restrictive than, but aligned with, the CAMARA community agreement on 2-legged vs. 3-legged access.

Context:
This issue was discussed here https://github.com/camaraproject/IdentityAndConsentManagement/issues/92 and I specifically mentioned something very similar to what you say above here https://github.com/camaraproject/IdentityAndConsentManagement/issues/92#issuecomment-1852448418. And then it was mentioned here https://github.com/camaraproject/IdentityAndConsentManagement/issues/92#issuecomment-1862432886 that if the WG only considers client credentials when no personal information is processed, that would be fine because it is indeed more restrictive. And then, in light of that condition, @eric-murray updated the documentation accordingly (PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/109), which refers to not requiring the purpose statement when client credentials is used (purpose must be always declared when personal information is processed as required by law).

Reply:
@jpengar @eric-murray I wasn't aware of this restrictive change done in a side discussion. For me the broadly discussed, reviewed and agreed clear [documented guideline](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#mandatory-template-for-infodescription-in-camara-api-specs) is still valid. The text is here very clear:

> Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.

So purpose for accessing the API can be already declared within the onboarding.

I would therefore ask to keep all documents in accordance with this agreement. It shouldn't be overruled by a PR reviewed and merged by one person.

Reply:
@hdamker
Well, not a "side discussion" as the issue and then PR submission followed the agreed procedures, and was formally approved during a [meeting](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/MeetingMinutes/2024/MOM-2024-01-17.md). And hardly even a "[discussion](https://github.com/camaraproject/IdentityAndConsentManagement/commit/7675e3e027b594cdf16bbced5374a131f0cc840e)". Including this additional statement was the price of getting an agreement on scope formation when using the client_credentials grant.

I'm not a fan of including such "narrative" text in specifications for precisely this reason. If you believe the added text contradicts statements made elsewhere in the document, then I'd be more than happy to revert to the original PR text.

Reply:
@eric-murray 
Looking on the [minutes](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/MeetingMinutes/2024/MOM-2024-01-17.md) I can't imagine that it was discussed in detail before the "agreement". Also the commit was done just a day before the meeting, so there was no sufficient time to review and assume a consensus. Given that the previous consensus which was changed here "on the side" (as this was not the subject of the issue and the PR) was discussed over weeks and reviewed by many people, I wouldn't consider this an appropriate way to change such consensus.

Yes, I will open an issue to revert to your original PR text, in line with our mandatory guideline.

Reply:
@eric-murray @hdamker 

I'm not going to go back over the procedures that were followed, or discuss comments like _"It shouldn't be overruled by a PR that was reviewed and merged by one person_ (thanks for remove it @hdamker) . I already find them quite disrespectful, so I'm pretty sure that everything will be done much better from now on. It looks like I was in different meetings and different discussions than you guys. I don't know :/

I'm absolutely fine to revisit this topic, but please note that the change requested in this PR with respect to the original text was because otherwise there could be scenarios where personal information is processed (with no opt-in or opt-out) where customer credentials would be allowed but the purpose is not declared. 

From Telefónica's point of view, the purpose must always be declared when processing personal data (this is what the GDPR requires). So, it's not just a matter of going back to Eric's original PR text, but also of reconsidering whether the purpose needs to be declared when client credentials are used but users' personal data is processed. IMHO, it needs to be declared in those cases, so Eric's original PR text would be against it.

Reply:
https://github.com/camaraproject/IdentityAndConsentManagement/issues/131 created for the statement introduced by #92. We shouldn't repeat here the same mistake.

Reply:
@jpengar 

> From Telefónica's point of view, the purpose must always be declared when processing personal data (this is what the GDPR requires). So, it's not just a matter of going back to Eric's original PR text, but also of reconsidering whether the purpose needs to be declared when client credentials is used but user's personal data is processed. IMHO, it needs to be declared in those cases, so Eric's original PR text would be against it.

There is now #131 which is better suited for this discussion. Would you like to repeat the argument there?

Reply:
I suppose this conversation can be set to resolved. @DT-DawidWroblewski @AxelNennker 

The commented text has been updated by https://github.com/camaraproject/IdentityAndConsentManagement/pull/121/commits/7f5f42f5271c74f94b15564bc1ebbed21a7803fe and is now in line with our [documented guideline](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#mandatory-template-for-infodescription-in-camara-api-specs).